plugin/backend_unsupported.go

```diff
@@ -24,8 +25,11 @@ func (pm *Manager) Enable(name string, config *types.PluginEnableConfig) error {
 }
 
 // Inspect examines a plugin config
-func (pm *Manager) Inspect(name string) (tp types.Plugin, err error) {
-	return tp, errNotSupported
+func (pm *Manager) Inspect(refOrID string) (tp types.Plugin, err error) {
+	// Even though plugin is not supported, we still want to return `not found`
+	// error so that `docker inspect` (without `--type` specified) returns correct
+	// `not found` message
+	return tp, fmt.Errorf("no such plugin name or ID associated with %q", refOrID)
```

Comment:
The correct fix would probably be to skip `errNotSupported` errors if no `--type` is specified on `docker inspect`; https://github.com/docker/docker/blob/24ffa2fa7751e2f1fbdd7e0d5a640f24e8a5b803/cli/command/system/inspect.go#L131-L133